lib/logger.py

```diff
@@ -96,7 +96,8 @@ def read_gen(self, only_stream_id=None):
             microseconds, stream_id, size = struct.unpack('IHH', header)
             dt = datetime.timedelta(microseconds=microseconds)
             data = self.f.read(size)
-            if only_stream_id is None or only_stream_id == stream_id:
+            if ((only_stream_id is None and multiple_streams == []) or
+                only_stream_id == stream_id or stream_id in multiple_streams):
```

Comment:
I would append `only_stream_id` to `multiple_streams` and convert `multiple_streams to `set`. The if would then be

```python
if len(multiple_streams) == 0 or stream_id in multiple_streams:
```